### PR TITLE
black: 21.12b0 -> 22.1.0

### PR DESCRIPTION
**Pull request by aaronjheng**
###### Motivation for this change

https://github.com/psf/black/releases/tag/22.1.0

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by sveitser — CHANGES_REQUESTED** [review]
Had a look at the dependencies and I think we don't need `attrs` anymore. https://github.com/psf/black/commit/31f4105731c9e5a6930905d358b62b2b834b940b

Apart from that LGTM, thanks!

File: pkgs/development/python-modules/black/default.nix [hunk 1]
```diff
@@ -1,5 +1,10 @@
...
+{ stdenv
+, lib
+, buildPythonPackage
+, fetchPypi
+, pythonOlder
+, setuptools-scm
+, pytestCheckHook
 , aiohttp
 , aiohttp-cors
 , attrs
```
Comment (sveitser) [inline pkgs/development/python-modules/black/default.nix:10]:
```suggestion
```

**Review by sveitser — APPROVED** [review]
LGTM

File: pkgs/development/python-modules/black/default.nix [hunk 2]
```diff
@@ -71,7 +74,7 @@ buildPythonPackage rec {
     typed-ast # required for tests and python2 extra
     uvloop
   ] ++ lib.optional (pythonOlder "3.7") dataclasses
```
Comment (dotlambda) [inline pkgs/development/python-modules/black/default.nix:76]:
```suggestion
  ]
```
We don't have Python 3.6 anymore.

File: pkgs/development/python-modules/black/default.nix [hunk 3]
```diff
@@ -71,7 +74,7 @@ buildPythonPackage rec {
...
-    ++ lib.optional (pythonOlder "3.8") typing-extensions;
+  ++ lib.optional (pythonOlder "3.8") typing-extensions;
```
Comment (dotlambda) [inline pkgs/development/python-modules/black/default.nix:77]: This looks wrong: https://github.com/psf/black/pull/2772

File: pkgs/development/python-modules/black/default.nix [hunk 4]
```diff
@@ -71,7 +74,7 @@ buildPythonPackage rec {
     typed-ast # required for tests and python2 extra
```
Comment (dotlambda) [inline pkgs/development/python-modules/black/default.nix:74]: only required for Python < 3.8

**Comment by dotlambda** [comment]
Result of `nixpkgs-review pr 157800` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>python3.pkgs.ipython</li>
  </ul>
</details>

**Comment by dotlambda** [comment]
Please target staging.

**Comment by aaronjheng** [comment]
@dotlambda You mean `staging` branch, right?

**Comment by aaronjheng** [comment]
@dotlambda Done.

**Pull request merged**